### PR TITLE
bug(DataList): remove multiple select support

### LINKED ISSUE #7688: Data list- selectable variants should only be single select

### PR DESCRIPTION
**Pull request by wise-king-sullyman**
**What**: Closes #7688 

**Additional issues**:
Related to #7491

### CODE REVIEW

**Review by thatblindgeye — APPROVED** [review]

File: packages/react-core/src/components/DataList/DataList.tsx [hunk 1]
```diff
@@ -21,8 +21,6 @@ export enum DataListWrapModifier {
 }
 
 export interface SelectableRowObject {
-  /** Determines if only one of the selectable rows should be selectable at a time */
-  type: 'multiple' | 'single';
```
Comment (nicolethoen) [inline packages/react-core/src/components/DataList/DataList.tsx:25]: This feels like a breaking change. Maybe we can just console log a warning if they use 'multiple' as the value and make a note in the prop description that it's always single?

Comment (wise-king-sullyman) [reply packages/react-core/src/components/DataList/DataList.tsx:25]: I'm fine with doing that instead, but I thought we could make breaking changes to beta props?

Comment (nicolethoen) [reply packages/react-core/src/components/DataList/DataList.tsx:25]: op - you're right. I'm fine with it then :)

**Review by nicolethoen — APPROVED** [review]

**Review by mmenestr — APPROVED** [review]
LGTM!

**Pull request merged**